src/dolphin/filtering.py

```diff
@@ -0,0 +1,198 @@
+import numpy as np
+from numpy.typing import ArrayLike
+from scipy import ndimage
+
+
+def filtering(
+    unw_ifg: ArrayLike,
+    corr: ArrayLike,
+    mask_cutoff: float = 0.5,
+    wavelength_cutoff: float = 50 * 1e3,
+    dx: float = 30,
+) -> np.ndarray:
+    """Filter out signals with spatial wavelength longer than a threshold.
+
+    Parameters
+    ----------
+    unw_ifg : np.ndarray, 2D complex array
+        unwrapped interferogram to interpolate
+    corr : 2D float array
+        Array of interferometric correlation from 0 to 1.
+    mask_cutoff: float
+        Threshold to use on `corr` so that pixels where
+        `corr[i, j] > mask_cutoff` are True and the rest are False.
+        The default is 0.5.
+    wavelength_cutoff: float
+        Spatial wavelength threshold to filter unw_ifg.
+        Signals with wavelength longer than 'wavelength_cutoff' in unw_ifg will be
+        filtered out.
+        The default is 50*1e3 (m).
+    dx : float
+        Pixel spatial spacing. Assume same spacing for x, y axes.
+        The default is 30 (m).
+
+
+    Returns
+    -------
+    filtered_ifg : 2D complex array
+        filtered interferogram that does not contain signals with spatial wavelength
+        longer than a threshold.
+
+    """
+    nrow, ncol = corr.shape
+
+    # Create Boolean mask for corr > mask_cutoff to be True and the rest to be False
+    mask = (corr > mask_cutoff).astype("bool")
+
+    # Create Boolean mask for Zero-filled boundary area to be False
+    # and the rest to be True
+    mask_boundary = ~(corr == 0).astype("bool")
+
+    # Ramp plane fitting
+    plane = fit_ramp_plane(unw_ifg, mask)
+
+    # Replace masked out pixels with the ramp plane
+    unw_ifg_interp = np.copy(unw_ifg)
```

Comment:
```suggestion
    plane = fit_ramp_plane(unwrapped_phase, mask)

    # Replace masked out pixels with the ramp plane
    unw_ifg_interp = np.copy(unwrapped_phase)
```
(trying to catch all my name changes)